### ISSUE TITLE
[FEAT]: Allow Custom Configuration for File Extensions

### ISSUE THREAD

**Issue by xewelus** [Mintplex-Labs/anything-llm#2638]
### What would you like to see?

I use LINQPad for programming. It uses `.linq` files, which are essentially `.cs` files with additional metadata. Unfortunately, these file types are not recognized as plain text. It would be great to have the ability to configure custom file extensions. I checked the source code and found that it relies on logic from the `mime` module, but there’s no way to adjust it flexibly.

**Comment by timothycarambat** [comment]
Agreed, manually maintaining that list has been a chore and we should abstract it to be able to load in a file as well that can be easily extended via convfig

**Comment by forgot** [comment]
This would definitely be a lot easier than having to fork/PR for every file type. I'm assuming the ability to add filetypes would have to be granular, with admins managing the list for business/organization/multi-user deployments and users at the app/single-user level?

I'm fairly new to the project, and willing to help out where I can. Are discussions around implementation details for features like this generally kept within the issue, or are they more of a Discord thing?

**Comment by timothycarambat** [comment]
Closing this since https://github.com/Mintplex-Labs/anything-llm/pull/2664 seems to have resolved the primary issue - which was parsing regular text-files. Since any binary format would require a code implementation like we do for PDFs, DocX, etc. 

Something that would be impossible to resolve with a list of supported extensions 👍
